### PR TITLE
add support for ratios

### PR DESCRIPTION
**Pull request by nxvl**

### CODE REVIEW

**Comment by smathieu** [comment]
Nice! Trying this out now :)

**Comment by nxvl** [comment]
Tried with both area and line graph's for the ratios and line made way more sense, but it's a one liner required to change that.

File: app/assets/javascripts/prosperity/graph.js.coffee [hunk 1]
```diff
@@ -66,8 +66,18 @@ class SubGraph
...
+      @calculateRatios()
     @chart = new @class()(@chartOptions)
 
+  calculateRatios: ->
+    for item, index in @data
+      sum = 0
+      for key, value of item
+        sum += value unless key == 'x'
+      for key, value of item
+        item[key] = (value / sum) * 100 unless key == 'x'
```
Comment (smathieu) [inline app/assets/javascripts/prosperity/graph.js.coffee]: Possible division by 0 here.

Comment (smathieu) [inline app/assets/javascripts/prosperity/graph.js.coffee]: Getting the following stack trace. Might be related to this...

```
 Error: Invalid value for <path> attribute d="V25"raphael-min-2.1.2.js?body=1:10 qraphael-min-2.1.2.js?body=1:10 wraphael-min-2.1.2.js?body=1:11 c._engine.pathraphael-min-2.1.2.js?body=1:10 v.pathmorris.min-0.5.js?body=1:7 b.Grid.d.drawEventmorris.min-0.5.js?body=1:7 b.Grid.d.drawEventsmorris.min-0.5.js?body=1:7 b.Grid.d.redrawmorris.min-0.5.js?body=1:7 b.Grid.d.setDatamorris.min-0.5.js?body=1:7 dmorris.min-0.5.js?body=1:7 cmorris.min-0.5.js?body=1:7 cgraph.js?body=1:88 SubGraph.drawgraph.js?body=1:158 _results.push.$.ajax.successjquery.js?body=1:975 firejquery.js?body=1:1085 self.fireWithjquery.js?body=1:7804 donejquery.js?body=1:8519 callback
```

Comment (nxvl) [inline app/assets/javascripts/prosperity/graph.js.coffee]: fixed division by 0

Comment (nxvl) [inline app/assets/javascripts/prosperity/graph.js.coffee]: i'm not getting the stack trace on the console, can you confirm it's still there or it was fixed by the division by 0 patch?

Comment (smathieu) [inline app/assets/javascripts/prosperity/graph.js.coffee]: Still getting those. Looking at the the debugger, there are some NaN values in being returned:

https://www.evernote.com/shard/s78/sh/c8c88720-c388-4a68-adea-05d7486a85f4/5fd1b585e84f156523d1a61d649470ff

Comment (nxvl) [inline app/assets/javascripts/prosperity/graph.js.coffee]: on chrome works without issues, maybe is something about that specific data set? 
https://www.dropbox.com/s/0tb1frw2o3rr8jj/Screen%20Shot%202014-12-15%20at%2020.01.46?dl=0

**Pull request merged**